File: DONGUK/프로그래머스_WEEK_10/스킬트리.py
from collections import deque

def solution(skill, skill_trees):
    answer = 0
    d = {}
    
    for i in range(len(skill)):
        d[skill[i]] = i
    
    for skill in skill_trees:
        s_ind = []
        
        for i in range(len(skill)):
            if skill[i] in d:
                s_ind.append(d[skill[i]])
            
        
        flag = True
        store = []
        for ind in s_ind:
            if ind == 0:
                store.append(ind)
            
            else:
                if len(store) == 0 or store[-1] != ind-1:
                    flag = False
                    break
                    
                else:
                    store.append(ind)
        
        if flag:
            answer += 1
        
    return answer

File: JUNHEE/프로그래머스_week10/스킬트리.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string skill, vector<string> skill_trees) {
    int answer = 0;
    for(int i = 0; i < skill_trees.size(); i++){
        int idx = 0;
        bool flag = false;
        for(int j = 0; j < skill_trees[i].size(); j++){
            if(skill_trees[i][j] == skill[idx]){
                idx++;
                continue;
            }
            for(int k = idx + 1; k < skill.size(); k++){
                if(skill_trees[i][j] == skill[k]){
                    flag = true;
                }
            }
        }
        if(!flag){
            answer++;
        }
    }
    return answer;
}
